include/deal.II/fe/fe_tools.templates.h

```diff
@@ -3243,6 +3243,106 @@ namespace FETools
     return Utilities::invert_permutation(
       hierarchic_to_lexicographic_numbering<dim>(degree));
   }
+
+
+
+  template <int dim>
+  std::pair<std::vector<unsigned int>, std::vector<unsigned int>>
+  cell_to_face_patch(const unsigned int &degree,
+                     const unsigned int &direction,
+                     const bool         &cell_hierarchical_numbering,
+                     const bool         &is_continuous)
+  {
+    AssertThrow(dim > 1, ExcInvalidFEDimension(1, dim));
+    AssertThrow(direction < dim, ExcInvalidFEDimension(direction, dim));
+
+    // n_1d: DoFs per 1D edge (parallel to face)
+    const unsigned int n_1d = degree + 1;
+    // n_ort: DoFs in the direction orthogonal to the face in the combined face
+    // space For discontinuous: 2*n (n for cell0 face, n for cell1 face) For
+    // continuous: 2*n - 1 (shared DoFs)
+    const unsigned int n_ort_1d = 2 * n_1d - (is_continuous ? 1 : 0);
+
+    const unsigned int dofs_per_cell = Utilities::fixed_power<dim>(n_1d);
+    const unsigned int dofs_per_face_patch =
+      Utilities::fixed_power<dim - 1>(n_1d) * n_ort_1d;
+
+    std::array<unsigned int, dim> face_patch_dofs_in_direction;
+    for (unsigned int d = 0; d < dim; ++d)
+      face_patch_dofs_in_direction[d] = n_1d;
+    face_patch_dofs_in_direction[direction] = n_ort_1d;
+
+
+
```

Comment:
Let us use a single empty line within function, as the three lines are used between functions according to the style guide https://www.dealii.org/current/doxygen/deal.II/CodingConventions.html
```suggestion
```